test/native/demangleTest.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "arch.h"
+#include "testRunner.hpp"
+#include "demangle.h"
+#include <stdio.h>
+
+TEST_CASE(Demangle_test_needs_demangling) {
+    // Rust legacy-mangled symbol
+    CHECK_EQ(Demangle::needsDemangling("_ZN12panic_unwind3imp5panic17exception_cleanup17he4cf772173d90f46E"), true);
+    // Rust v0-mangled symbol
+    CHECK_EQ(Demangle::needsDemangling("_RNvCs6KtT2fMGqXk_8infiloop4main"), true);
+    // C++ symbol
+    CHECK_EQ(Demangle::needsDemangling("_ZNKSbIwSt11char_traitsIwESaIwEE4_Rep12_M_is_sharedEv"), true);
+    // C symbols
+    CHECK_EQ(Demangle::needsDemangling("malloc"), false);
+    CHECK_EQ(Demangle::needsDemangling("_malloc"), false);
+}
+
+// demangle calls _cxa_demangle, which should support demangling Itanium-mangled symbols across all Linux systems,
+// but might have a different mangling format on some other platforms.
+#ifdef __linux__
+TEST_CASE(Demangle_test_demangle_cpp) {
+    const char *s = Demangle::demangle("_ZNSt15basic_streambufIwSt11char_traitsIwEE9pbackfailEj", false);
+    CHECK_EQ (strcmp(s, "std::basic_streambuf<wchar_t, std::char_traits<wchar_t> >::pbackfail"), 0);
```

Comment:
`CHECK_EQ` supports comparing strings - no need for `strcmp`.
Also, redundant space before `(`.

Reply:
It didn't work for me for some reason and I tried it a few different ways. Do you have an example?

Reply:
An example is https://github.com/async-profiler/async-profiler/blob/f8e887eb9dcb998ca08c4757a2dbcae9395d9733/test/native/perfEventsLinuxTest.cpp#L16

Reply:
@krk 

I get errors like

```
        actual values: s = 94904844431968 (0x5650C18A3260), "std::basic_streambuf<wchar_t, std::char_traits<wchar_t> >::pbackfail" = 94904675958744 (0x5650B77F7FD8)
```

Reply:
It does work if you do it like this:
```
    char *s = Demangle::demangle("_ZNSt15basic_streambufIwSt11char_traitsIwEE9pbackfailEj", false);
    const char *e = "std::basic_streambuf<wchar_t, std::char_traits<wchar_t> >::pbackfail";
    CHECK_EQ(s, e);
```

Reply:
Trying to repro, I get this:

```
    const char *a = "asd";
    const char *b = "asdf";
    CHECK_EQ(a, b);

// Assertion failed: (a == b),
//         actual values: a = "asd", b = "asdf"
```

Could you share the code for the useless errors you got?